Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/jvm/CommandLine.kt

```diff
@@ -16,97 +16,53 @@
-    var headerFilterPrefix: Array<String> = arrayOf()
-
-    @Argument(value = "-compilerOpts", shortName = "-copt", valueDescription = "<arg>", description = "additional compiler options", delimiter = " ")
-    var compilerOpts: Array<String> = arrayOf()
-
-    @Argument(value = "-linkerOpts", shortName = "-lopt", valueDescription = "<arg>", description = "additional linker options", delimiter = " ")
-    var linkerOpts: Array<String> = arrayOf()
-
-    @Argument(value = "-shims", description = "wrap bindings by a tracing layer") 
-    var shims: Boolean = false
-
-    @Argument(value = "-linker", valueDescription = "<file>", description = "use specified linker") 
-
-    var linker: String? = null
-    @Argument(value = "-cstubsname", valueDescription = "<name>", description = "provide a name for the generated c stubs file") 
-    var cstubsname: String? = null
+fun getCommonInteropArguments() = listOf(
+        OptionDescriptor(ArgType.Boolean(), "verbose", description = "Enable verbose logging output", defaultValue = "false"),
+        OptionDescriptor(ArgType.Choice(listOf("jvm", "native", "wasm")),
+                "flavor", description = "Interop target", defaultValue = "jvm"),
+        OptionDescriptor(ArgType.String(), "pkg", description = "place generated bindings to the package"),
+        OptionDescriptor(ArgType.String(), "generated", description = "place generated bindings to the directory",
+                defaultValue = System.getProperty("user.dir")),
+        OptionDescriptor(ArgType.String(), "libraryPath", description = "add a library search path",
+                isMultiple = true, delimiter = ","),
+        OptionDescriptor(ArgType.String(), "manifest", description = "library manifest addend"),
+        OptionDescriptor(ArgType.String(), "natives", description = "where to put the built native files",
+                defaultValue = System.getProperty("user.dir")),
+        OptionDescriptor(ArgType.String(), "staticLibrary", description = "embed static library to the result",
+                isMultiple = true, delimiter = ","),
+        OptionDescriptor(ArgType.String(), "temporaryFilesDir", description = "save temporary files to the given directory")
+    )
+
+fun getCInteropArguments(): List<OptionDescriptor> {
+    val options = listOf(
+            OptionDescriptor(ArgType.String(), "import", description = "a semicolon separated list of headers, prepended with the package name",
+                    isMultiple = true, delimiter = ";"),
+            OptionDescriptor(ArgType.String(), "target", description = "native target to compile to"),
+            OptionDescriptor(ArgType.String(), "def", description = "the library definition file"),
+            OptionDescriptor(ArgType.String(), "header", "hd", "header file to produce kotlin bindings for",
+                    isMultiple = true, delimiter = ","),
```

Comment:
What's the motivation behind adding `,` delimiter to this and other options?

Reply:
This allows to save old behaviour when option can be splitted with some delimiter and get all values already splitted.

Reply:
Now I see that this appears to be a (weird) default behaviour of `CommonToolArguments` machinery.
I'm not sure this was the intended behaviour for `cinterop`, but I guess it is better to keep it anyway.

Reply:
It doesn't worked for interop tools. I can add them and resend to compiler.